packages/aws-cdk/lib/api/cxapp/stacks.ts

```diff
@@ -403,3 +412,39 @@ export interface Tag {
   readonly Key: string;
   readonly Value: string;
 }
+
+function _makeCdkMetadataAvailableCondition() {
+  const options = new Array<any>();
+  // Find out all regions known to support AWS::CDK::Metadata from the RegionInfo facts database
+  for (const region of regionInfo.Fact.regions) {
+    if (regionInfo.Fact.find(region, regionInfo.FactName.CDK_METADATA_RESOURCE_AVAILABLE) === 'YES') {
+      options.push({ 'Fn::Equals': [{ Ref: 'AWS::Region' }, region] });
+    }
+  }
+  return _fnOr(options);
+}
+
+/**
+ * This takes a bunch of operands and crafts an `Fn::Or` for those. Funny thing is `Fn::Or` requires
+ * at least 2 operands and at most 10 operands, so we have to... do this.
+ */
+function _fnOr(operands: any[]): any {
```

Comment:
I wonder if we can do something more generally reusable such as turning a fact into a `Mapping` and then using `Fn::FindInMap`. I guess the big question is how `Fn::FindInMap` will behave if the key is not found, we can't have it fail...

Reply:
You cannot do this to maps... Maps can only have keys 2 levels deep (and actually MUST have that).